### ISSUE TITLE
Successfully built x265 multi-bit depth but....

### ISSUE THREAD

**Issue by Lazinezz** [rdp/ffmpeg-windows-build-helpers#209]
Hi,
   I managed to succesfully build multi-bit depth x265 using ur build script,
x265 [info]: HEVC encoder version 2.3+17-6e348252e902
x265 [info]: build info [Windows][GCC 6.3.0][64 bit] 8bit+10bit+12bit

,but my apologies cos i had to modify ur script abit.If u dont mind, could you help to simplify it or do what is needed,i always use ur script to build ffmpeg cos its the best so far(atleast to me hehs) so here is what i modified in ur script..

build_libx265() {
  # the only one that uses mercurial, so there's some extra initial junk in this method... XXX needs some cleanup :|
  local checkout_dir=x265
  if [[ $high_bitdepth == "y" ]]; then
    checkout_dir=x265_high_bitdepth_10
  fi

  if [[ $prefer_stable = "n" ]]; then
    local old_hg_version
    if [[ -d $checkout_dir ]]; then
      cd $checkout_dir
      if [[ $git_get_latest = "y" ]]; then
        echo "doing hg pull -u x265"
        old_hg_version=`hg --debug id -i`
        hg pull -u || exit 1
        hg update || exit 1 # guess you need this too if no new changes are brought down [what the...]
      else
        echo "not doing hg pull x265"
        old_hg_version=`hg --debug id -i`
      fi
    else
      echo "doing hg clone x265"
      hg clone https://bitbucket.org/multicoreware/x265 $checkout_dir || exit 1
      cd $checkout_dir
      old_hg_version=none-yet
    fi
    cd source

    # hg checkout 9b0c9b # no longer needed, but once was...left here so I know how :)

    local new_hg_version=`hg --debug id -i`  
    if [[ "$old_hg_version" != "$new_hg_version" ]]; then
      echo "got upstream hg changes, forcing rebuild...x265"
      rm -f already*
    else
      echo "still at hg $new_hg_version x265"
    fi
  else
    # prefer_stable == "y" TODO clean this up...
    local old_hg_version
    if [[ -d $checkout_dir ]]; then
      cd $checkout_dir
      if [[ $git_get_latest = "y" ]]; then
        echo "doing hg pull -u x265"
        old_hg_version=`hg --debug id -i`
        hg pull -u || exit 1
        hg update || exit 1 # guess you need this too if no new changes are brought down [what the...]
      else
        echo "not doing hg pull x265"
        old_hg_version=`hg --debug id -i`
      fi
    else
      echo "doing hg clone x265"
      hg clone https://bitbucket.org/multicoreware/x265 $checkout_dir || exit 1
      cd $checkout_dir
      old_hg_version=none-yet
      create_build_dir
      local build_root
      build_root="$(pwd)"
      mkdir -p {8,10,12}bit
    fi
    cd 12bit

    # hg checkout 9b0c9b # no longer needed, but once was...

    local new_hg_version=`hg --debug id -i`  
    if [[ "$old_hg_version" != "$new_hg_version" ]]; then
      echo "got upstream hg changes, forcing rebuild...x265"
      rm -f already*
    else
      echo "still at hg $new_hg_version x265"
    fi
  fi
  
  local cmake_params="-DENABLE_SHARED=OFF"
  if [[ $high_bitdepth == "y" ]]; then
    cmake_params="$cmake_params -DHIGH_BIT_DEPTH=ON" # Enable 10 bits (main10) and 12 bits (???) per pixels profiles.
    if [ "$bits_target" = "32" ]; then
      cmake_params="$cmake_params -DENABLE_ASSEMBLY=OFF" # apparently required or build fails
    fi
  fi

  #if [ "$bits_target" = "32" ]; then
    cmake_params="$cmake_params -DWINXP_SUPPORT:BOOL=TRUE" # enable windows xp support apparently
  #fi

  do_cmake "$cmake_params ../source -DHIGH_BIT_DEPTH:BOOL=ON -DMAIN12:BOOL=ON -DENABLE_SHARED:BOOL=OFF -DEXPORT_C_API:BOOL=OFF -DENABLE_CLI:BOOL=OFF"
  do_make
  cp libx265.a ../8bit/libx265_main12.a
  cd ..

  cd 10bit
  do_cmake "$cmake_params ../source -DHIGH_BIT_DEPTH:BOOL=ON -DMAIN10:BOOL=ON -DENABLE_SHARED:BOOL=OFF -DEXPORT_C_API:BOOL=OFF -DENABLE_CLI:BOOL=OFF"
  do_make
  cp libx265.a ../8bit/libx265_main10.a
  cd ..

  cd 8bit
  do_cmake "$cmake_params ../source -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT:BOOL=ON -DLINKED_12BIT:BOOL=ON -DEXTRA_LIB=libx265_main10.a;libx265_main12.a"
  do_make
  mv libx265.a libx265_main.a
  ar -M <<EOF
CREATE libx265.a
ADDLIB libx265_main.a
ADDLIB libx265_main10.a
ADDLIB libx265_main12.a
SAVE
END
EOF
  do_make_install
  cd ../..
}

thanks so much.

**Comment by rdp** [comment]
Could you send me a diff of that and the original?

**Comment by rzcat** [comment]
wow!

**Comment by Lazinezz** [comment]
im sorry i dont know how to send you(lol im ashamed,imma newbie lol), original is yours, modified is my first post.

**Comment by Lazinezz** [comment]
send you a text of the modified,as im not a linux techie, i dont know how to diff.
[Modified Cross Compile Script.txt](https://github.com/rdp/ffmpeg-windows-build-helpers/files/812512/Modified.Cross.Compile.Script.txt)

**Comment by Lazinezz** [comment]
rzcat,
i compiled ffmpeg using gcc 6.3.0 in linux vm, n by using rdp's build script both multi-bitdepth of x265 and ffmpeg with multi-bitdepth libx265 were created in x265 folder and x86_64-mingw64 bin folder respectively.

**Comment by rzcat** [comment]
Well, the modified script doesnt work on my Fedora, stuck at here:
```shell
    [ 61%] Built target common
    [ 83%] Built target encoder
    [ 85%] Built target x265-static
    [ 86%] Linking CXX executable x265.exe
    libx265.a: error adding symbols: Archive has no index; run ranlib to add one
    collect2: error: ld returned 1 exit status
    CMakeFiles/cli.dir/build.make:360: recipe for target 'x265.exe' failed
    make[2]: *** [x265.exe] Error 1
    CMakeFiles/Makefile2:137: recipe for target 'CMakeFiles/cli.dir/all' failed
    make[1]: *** [CMakeFiles/cli.dir/all] Error 2
    Makefile:127: recipe for target 'all' failed
    make: *** [all] Error 2
```

**Comment by Lazinezz** [comment]
hmm,
strange..i got no problems compiling x265, correct me if im wrong...that error has got to do with " ar " right? CMAKE_AR_PATH

**Comment by rzcat** [comment]
it's getting weird, i took a 2nd and a 3rd shot, both stuck at here but different from 1st time:
```shell
  CC     build/SDL_dummyaudio.lo
  CC     build/SDL_windows.lo
  CC     build/SDL_xinput.lo
In file included from src/core/windows/SDL_xinput.c:24:0:
src/core/windows/SDL_xinput.h:118:5: error: unknown type name 'XINPUT_GAMEPAD_EX'
     XINPUT_GAMEPAD_EX Gamepad;
     ^~~~~~~~~~~~~~~~~
src/core/windows/SDL_xinput.h:131:5: error: unknown type name 'XINPUT_STATE_EX'
     XINPUT_STATE_EX* pState     /* [out] Receives the current state */
     ^~~~~~~~~~~~~~~
src/core/windows/SDL_xinput.h:157:8: error: unknown type name 'XInputGetState_t'
 extern XInputGetState_t SDL_XInputGetState;
        ^~~~~~~~~~~~~~~~
src/core/windows/SDL_xinput.c:29:1: error: unknown type name 'XInputGetState_t'
 XInputGetState_t SDL_XInputGetState = NULL;
 ^~~~~~~~~~~~~~~~
In file included from ./include/SDL_platform.h:166:0,
                 from ./include/SDL_config.h:33,
                 from src/core/windows/../../SDL_internal.h:34,
                 from src/core/windows/SDL_xinput.c:21:
./include/begin_code.h:143:14: warning: initialization makes integer from pointer without a cast  -Wint-conversion]
 #define NULL ((void *)0)
              ^
src/core/windows/SDL_xinput.c:29:39: note: in expansion of macro 'NULL'
 XInputGetState_t SDL_XInputGetState = NULL;
                                       ^~~~
src/core/windows/SDL_xinput.c: In function 'WIN_LoadXInputDLL':
src/core/windows/SDL_xinput.c:107:27: error: 'XInputGetState_t' undeclared (first use in this function)
     SDL_XInputGetState = (XInputGetState_t)GetProcAddress((HMODULE)s_pXInputDLL, (LPCSTR)100);
                           ^~~~~~~~~~~~~~~~
src/core/windows/SDL_xinput.c:107:27: note: each undeclared identifier is reported only once for each function it appears in
src/core/windows/SDL_xinput.c:107:44: error: expected ';' before 'GetProcAddress'
     SDL_XInputGetState = (XInputGetState_t)GetProcAddress((HMODULE)s_pXInputDLL, (LPCSTR)100);
                                            ^~~~~~~~~~~~~~
src/core/windows/SDL_xinput.c:109:48: error: expected ';' before 'GetProcAddress'
         SDL_XInputGetState = (XInputGetState_t)GetProcAddress((HMODULE)s_pXInputDLL, "XInputGetState");
                                                ^~~~~~~~~~~~~~
Makefile:547: recipe for target 'build/SDL_xinput.lo' failed
make: *** [build/SDL_xinput.lo] Error 1
make: *** Waiting for unfinished jobs....
```

**Comment by Lazinezz** [comment]
That's a different issue already...look up  ffmpeg-windows-build-helpers/patches/sdl2.xinput.diff...to resolve that u have manually edit SDL_xinput.h or remove the " apply patch " line in rdp's script. I had the same prob too, maually edited and its all smooth, good luck.

**Comment by rzcat** [comment]
thanks for the quick respose, https://github.com/rdp/ffmpeg-windows-build-helpers/issues/207 this one?
weird it is that i just ran the original repo's script using gcc 4.9.4 inside the script and everything went well to sucess, il take a shot about commenting out the "apply patch" line later

**Comment by Lazinezz** [comment]
it's fine with gcc 4.94 but not later..i think thats why rdp stick with gcc 4.x and not the latest untill these issues are solved....as im impatient, i have to manual edits if i wanna use latest gcc, n now i already compiled ffmpeg with gcc 7.0.1...till then, will wait for rdp to resolve all issues as to me his build script is the best.

**Comment by rzcat** [comment]
Took another shot, modified script with **line 1169** commented out, again lead to the link error above
And I also tried to use @rdp 's original script, using --mingw-w64-ver=**5.0.1** --gcc-ver=**6.3.0** with line 1146 commented out, this time it went **well** with building all-gcc(Fedora 25 gcc6.3.1), and **bypass** the patching sdl2.xinput.diff error(because it's commented out), and finally stuck at
```shell
Already up-to-date.
got no code changes, not forcing reconfigure for that...
configuring ffmpeg_git (/ff_comp_test/sandbox/x86_64/ffmpeg_git) as $ PKG_CONFIG_PATH=/ff_comp_test/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/lib/pkgconfig PATH=/ff_comp_test/sandbox/cross_compilers/mingw-w64-x86_64/bin:$PATH ./configure --arch=x86_64 --target-os=mingw32 --cross-prefix=/ff_comp_test/sandbox/cross_compilers/mingw-w64-x86_64/bin/x86_64-w64-mingw32- --pkg-config=pkg-config --disable-w32threads --enable-libsoxr --enable-fontconfig --enable-libass --enable-libbluray --enable-iconv --enable-libtwolame --extra-cflags=-DLIBTWOLAME_STATIC --enable-libzvbi --enable-libcaca --enable-libmodplug --extra-libs=-lstdc++ --extra-libs=-lpng --enable-decklink --extra-libs=-loleaut32  --enable-libmp3lame --enable-version3 --enable-zlib --enable-librtmp --enable-libvorbis --enable-libtheora --enable-libspeex --enable-libopenjpeg --enable-gnutls --enable-libgsm --enable-libfreetype --enable-libopus --enable-bzlib --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-libschroedinger --enable-libvpx --enable-libilbc --enable-libwavpack --enable-libwebp --enable-libgme --enable-dxva2 --enable-avisynth --enable-gray --enable-libopenh264 --enable-netcdf  --enable-libflite --enable-lzma --enable-libsnappy --enable-libzimg --enable-gpl --enable-libx264 --enable-libx265 --enable-frei0r --enable-filter=frei0r --enable-librubberband --enable-libvidstab --enable-libxavs --enable-libxvid --enable-libmfx --enable-avresample --extra-libs=-lpsapi --extra-libs=-lspeexdsp --extra-cflags=-mtune=core2 --extra-cflags=-O3 --enable-static --disable-shared  --prefix=/ff_comp_test/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32 --enable-runtime-cpudetect
**ERROR: rubberband not found using pkg-config**
```
i ran twice, same result. And I found out that using --mingw-w64-ver=**5.0.1** is **way much faster** than using --mingw-w64-ver=**git**
To sum up what ive done these days,
@rdp 's original script
1. --mingw-w64-ver=**git** --gcc-ver=**4.9.3** => Building all-gcc... error(posted before)
2. --mingw-w64-ver=**git** --gcc-ver=**4.9.4** => Sucess
3. --mingw-w64-ver=**5.0.1** --gcc-ver=**6.3.0** => Patching sdl2.xinput.diff error(posted above)
4. --mingw-w64-ver=**5.0.1** --gcc-ver=**6.3.0** with **line 1146** commented out => ERROR: rubberband not found using pkg-config(posted above)

@Lazinezz 's modified script
1. --mingw-w64-ver=**5.0.1** --gcc-ver=**6.3.0** => Patching sdl2.xinput.diff error
2. --mingw-w64-ver=**5.0.1** --gcc-ver=**6.3.0** with **line 1169** commented out => [ 86%] Linking CXX executable x265.exe ERROR(posted above)
3. --mingw-w64-ver=**git** --gcc-ver=**6.3.0** with **line 1169** commented out => [ 86%] Linking CXX executable x265.exe ERROR(posted above)

**Comment by Lazinezz** [comment]
youre one step to success lol...about that rubberband not found using pkg-config, look at this link https://github.com/rdp/ffmpeg-windows-build-helpers/issues/203 ..its another issue with ffmpeg, try deleting the image base line or schmidthubert's workaround. its an issue for gcc 6.x onwards.

**Comment by rdp** [comment]
I've adjusted the script to use mingw-w64 5.0.1 now, thanks for the hint!

**Comment by rdp** [comment]
does this create x265.exe with 


x265 [info]: build info [Windows][GCC 6.3.0][64 bit] 8bit+10bit+12bit

or where is that output coming from?  Who is "using" this with all those bit depths?

**Comment by Lazinezz** [comment]
yes it creates x265.exe with those bit depths and are used by those who prefer to encode their own video files. 10bit depth produce better quality n compression etc but requires more cpu power n its getting more popular now.

**Comment by rzcat** [comment]
@Lazinezz Replacing that line with yours still leads to error
```shell
[ 61%] Built target common
[ 83%] Built target encoder
[ 85%] Built target x265-static
[ 86%] Linking CXX executable x265.exe
libx265.a: error adding symbols: Archive has no index; run ranlib to add one
collect2: error: ld returned 1 exit status
CMakeFiles/cli.dir/build.make:360: recipe for target 'x265.exe' failed
make[2]: *** [x265.exe] Error 1
CMakeFiles/Makefile2:137: recipe for target 'CMakeFiles/cli.dir/all' failed
make[1]: *** [CMakeFiles/cli.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

@rdp there're tons of ott android box here in China, and they are really cheap, with the market growing rapidly. the cpu of mine is Amlogic S905, this can decode hevc x265 10bit up to 4k, and the price is about RMB200+(about USD30+) for a S905/S905X box, with 2G/16G+WiFi(2.4/5)+Bluetooth+GigabitLan....(well it may sound like an ad -_-)
And the compression, size of a video encoded in x265 10bit is just about half the size of x264 based on the same picture quality i could accept, that can save **A LOT** HDD space. Having a multi-bitdepth-lib x265 encoder would be handy in different situations.
http://www.msystem.waw.pl/x265/ has 8+10+12 multilib versions of x265 already. to be frank, i only need 8+10 for now.

**Comment by Lazinezz** [comment]
hmm strange, ive just finished building x265 of 8+10bits with no errors,
x265 [info]: HEVC encoder version 2.3+17-6e348252e902
x265 [info]: build info [Windows][GCC 7.0.1][64 bit] 8bit+10bit  with the attached settings
[8bit n 10bit only.txt](https://github.com/rdp/ffmpeg-windows-build-helpers/files/833089/8bit.n.10bit.only.txt) , u might want to compare it with urs, hope u get it right this time

**Comment by rzcat** [comment]
@Lazinezz Using --mingw-w64-ver=5.0.1 --gcc-ver=6.3.0, I've completely replace build_libx265() section with yours, but the result is the same as before. Thank you all the same, as I'll just have to use the one produced from the sucess attempt before for a while encoding some videos out to save some space.

**Comment by kgp700** [comment]
It's interesting but I got error

Ubuntu 14.04LTS
GCC 7.10

Lazinezz's Modified Cross Compile Script.txt

[ 35%] Building ASM_YASM object common/CMakeFiles/common.dir/x86/mc-a2.asm.obj
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/ssd-a.asm:146: error: undefined symbol `m8' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/ssd-a.asm:146: error:  (Each undefined symbol is reported only once.)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/ssd-a.asm:181: error: undefined symbol `m9' (first use)
make[2]: *** [common/CMakeFiles/common.dir/x86/ssd-a.asm.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 35%] Built target encoder
make[1]: *** [common/CMakeFiles/common.dir/all] Error 2
make: *** [all] Error 2

Lazinezz's 8bit n 10bit only.txt

[ 40%] Building ASM_YASM object common/CMakeFiles/common.dir/x86/pixeladd8.asm.obj
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/ssd-a.asm:146: error: undefined symbol `m8' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/ssd-a.asm:146: error:  (Each undefined symbol is reported only once.)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/ssd-a.asm:181: error: undefined symbol `m9' (first use)
make[2]: *** [common/CMakeFiles/common.dir/x86/ssd-a.asm.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 40%] Built target encoder
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13872: error: undefined symbol `r7' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13872: error:  (Each undefined symbol is reported only once.)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13873: error: undefined symbol `xmmm8' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13874: error: undefined symbol `xmmm11' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13874: error: undefined symbol `xmmm9' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13876: error: undefined symbol `m8' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13876: error: undefined symbol `m9' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13930: error: undefined symbol `m11' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13949: error: undefined symbol `m10' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13995: error: (HADDD:1) undefined symbol `sizeofm10' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13995: error: (HADDD:8) undefined symbol `sizeofm10' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:13997: error: undefined symbol `xmmm10' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14094: error: undefined symbol `m12' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14142: error: (HADDD:1) undefined symbol `sizeofm12' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14142: error: (HADDD:8) undefined symbol `sizeofm12' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14190: error: (HADDD:1) undefined symbol `sizeofm10' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14190: error: (HADDD:8) undefined symbol `sizeofm10' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14191: error: undefined symbol `xmmm12' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14279: error: undefined symbol `m13' (first use)
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14327: error: (HADDD:1) undefined symbol `sizeofm13' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14327: error: (HADDD:8) undefined symbol `sizeofm13' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14375: error: (HADDD:1) undefined symbol `sizeofm12' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14375: error: (HADDD:8) undefined symbol `sizeofm12' in preprocessor
/home/kgpbuild/ffmpegbuild5/sandbox/win32/x265/source/common/x86/pixel-a.asm:14376: error: undefined symbol `xmmm13' (first use)
make[2]: *** [common/CMakeFiles/common.dir/x86/pixel-a.asm.obj] Error 1
make[1]: *** [common/CMakeFiles/common.dir/all] Error 2
make: *** [all] Error 2

**Comment by rdp** [comment]
I "think" those mean that x265 is built using -sse something but then
linked without that flag or something weird [?]

**Comment by zplov** [comment]
I'm having the same errors as kgp700 when using Lazinezz's 8bit n 10bit only.txt.  Oddly enough, it looks like the 10bit version built, but the errors show up in the 8b (see output below). 

Any idea how to solve this?  

[error.txt](https://github.com/rdp/ffmpeg-windows-build-helpers/files/1205825/error.txt)




Starting 32-bit builds...
doing hg clone x265
applying clone bundle from https://media-api.atlassian.io/file/862cce98-b3ed-433e-9e0c-74ca819cdd70/binary?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiI5YWNjYTUyYy0yZTAyLTRmMGYtYWI5Ny05MzlhYjQ0OWIzNjciLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOjg2MmNjZTk4LWIzZWQtNDMzZS05ZTBjLTc0Y2E4MTljZGQ3MCI6WyJyZWFkIl19LCJuYmYiOjE1MDIxMjQ2NjcsImV4cCI6MTUwMjEyNTA4N30.5BSzd55YpFQ4iptIQ6TFL8lyhliJl8TpLJfKUy2hYhw&client=9acca52c-2e02-4f0f-ab97-939ab449b367
adding changesets
adding manifests
                                                                                                                                                                adding file changes
                                                                                                                                                                added 11730 changesets with 27413 changes to 745 files
finished applying clone bundle
searching for changes
adding changesets
                                                                                                                                                                adding manifests
adding file changes
added 138 changesets with 623 changes to 216 files
updating to branch default
                                                                                                                                                                283 files updated, 0 files merged, 0 files removed, 0 files unresolved
./cross_compile_ffmpeg.sh: line 1277: create_build_dir: command not found
got upstream hg changes, forcing rebuild...x265
doing cmake in /home/zplovani/sandbox/win32/x265_high_bitdepth_10/10bit with PATH=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH with extra_args=-DENABLE_SHARED=OFF -DHIGH_BIT_DEPTH=ON -DENABLE_ASSEMBLY=OFF -DWINXP_SUPPORT:BOOL=TRUE ../source -DHIGH_BIT_DEPTH:BOOL=ON -DMAIN10:BOOL=ON -DENABLE_SHARED:BOOL=OFF -DEXPORT_C_API:BOOL=OFF -DENABLE_CLI:BOOL=OFF like this:
cmake -GUnix Makefiles . -DENABLE_STATIC_RUNTIME=1 -DCMAKE_SYSTEM_NAME=Windows -DCMAKE_RANLIB=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-ranlib -DCMAKE_C_COMPILER=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-gcc -DCMAKE_CXX_COMPILER=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++ -DCMAKE_RC_COMPILER=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-windres -DCMAKE_INSTALL_PREFIX=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32 -DENABLE_SHARED=OFF -DHIGH_BIT_DEPTH=ON -DENABLE_ASSEMBLY=OFF -DWINXP_SUPPORT:BOOL=TRUE ../source -DHIGH_BIT_DEPTH:BOOL=ON -DMAIN10:BOOL=ON -DENABLE_SHARED:BOOL=OFF -DEXPORT_C_API:BOOL=OFF -DENABLE_CLI:BOOL=OFF
-- cmake version 3.6.2
-- The C compiler identification is GNU 7.1.0
-- The CXX compiler identification is GNU 7.1.0
-- Check for working C compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-gcc
-- Check for working C compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++
-- Check for working CXX compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detected x86 target processor
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Performing Test CC_HAS_NO_STRICT_OVERFLOW
-- Performing Test CC_HAS_NO_STRICT_OVERFLOW - Success
-- Performing Test CC_HAS_NO_NARROWING
-- Performing Test CC_HAS_NO_NARROWING - Success
-- Performing Test CC_HAS_NO_ARRAY_BOUNDS
-- Performing Test CC_HAS_NO_ARRAY_BOUNDS - Success
-- Performing Test CC_HAS_FAST_MATH
-- Performing Test CC_HAS_FAST_MATH - Success
-- Performing Test CC_HAS_STACK_REALIGN
-- Performing Test CC_HAS_STACK_REALIGN - Success
-- Performing Test CC_HAS_FNO_EXCEPTIONS_FLAG
-- Performing Test CC_HAS_FNO_EXCEPTIONS_FLAG - Success
-- Found yasm: /usr/bin/yasm.exe (found version "1.3.0")
-- Found Yasm 1.3.0 to build assembly primitives
-- hg found at /usr/bin/hg
-- x265 version 2.5+6-d11482e5fedb
-- Looking for strtok_r
-- Looking for strtok_r - found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    ENABLE_STATIC_RUNTIME
    MAIN10


-- Build files have been written to: /home/zplovani/sandbox/win32/x265_high_bitdepth_10/10bit

making /home/zplovani/sandbox/win32/x265_high_bitdepth_10/10bit as $ PATH=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH make  -j 4

Scanning dependencies of target encoder
Scanning dependencies of target common
[  4%] Building CXX object encoder/CMakeFiles/encoder.dir/analysis.cpp.obj[  4%]
Building CXX object encoder/CMakeFiles/encoder.dir/search.cpp.obj
[  6%] Building CXX object encoder/CMakeFiles/encoder.dir/bitcost.cpp.obj
[  8%] Building CXX object common/CMakeFiles/common.dir/winxp.cpp.obj
[ 10%] Building CXX object encoder/CMakeFiles/encoder.dir/motion.cpp.obj
[ 12%] Building CXX object common/CMakeFiles/common.dir/primitives.cpp.obj
[ 14%] Building CXX object common/CMakeFiles/common.dir/pixel.cpp.obj
[ 16%] Building CXX object encoder/CMakeFiles/encoder.dir/slicetype.cpp.obj
[ 18%] Building CXX object common/CMakeFiles/common.dir/dct.cpp.obj
[ 20%] Building CXX object encoder/CMakeFiles/encoder.dir/frameencoder.cpp.obj
[ 22%] Building CXX object common/CMakeFiles/common.dir/ipfilter.cpp.obj
[ 24%] Building CXX object encoder/CMakeFiles/encoder.dir/framefilter.cpp.obj
[ 26%] Building CXX object encoder/CMakeFiles/encoder.dir/level.cpp.obj
[ 28%] Building CXX object common/CMakeFiles/common.dir/intrapred.cpp.obj
[ 30%] Building CXX object common/CMakeFiles/common.dir/loopfilter.cpp.obj
[ 32%] Building CXX object encoder/CMakeFiles/encoder.dir/nal.cpp.obj
[ 34%] Building CXX object encoder/CMakeFiles/encoder.dir/sei.cpp.obj
[ 36%] Building CXX object common/CMakeFiles/common.dir/constants.cpp.obj
[ 38%] Building CXX object encoder/CMakeFiles/encoder.dir/sao.cpp.obj
[ 40%] Building CXX object encoder/CMakeFiles/encoder.dir/entropy.cpp.obj
[ 42%] Building CXX object common/CMakeFiles/common.dir/cpu.cpp.obj
[ 44%] Building CXX object common/CMakeFiles/common.dir/version.cpp.obj
[ 46%] Building CXX object encoder/CMakeFiles/encoder.dir/dpb.cpp.obj
[ 48%] Building CXX object common/CMakeFiles/common.dir/threading.cpp.obj
[ 50%] Building CXX object encoder/CMakeFiles/encoder.dir/ratecontrol.cpp.obj
[ 52%] Building CXX object common/CMakeFiles/common.dir/threadpool.cpp.obj
[ 54%] Building CXX object encoder/CMakeFiles/encoder.dir/reference.cpp.obj
[ 56%] Building CXX object common/CMakeFiles/common.dir/wavefront.cpp.obj
[ 58%] Building CXX object encoder/CMakeFiles/encoder.dir/encoder.cpp.obj
[ 60%] Building CXX object common/CMakeFiles/common.dir/md5.cpp.obj
[ 62%] Building CXX object common/CMakeFiles/common.dir/bitstream.cpp.obj
[ 64%] Building CXX object encoder/CMakeFiles/encoder.dir/api.cpp.obj
[ 66%] Building CXX object common/CMakeFiles/common.dir/yuv.cpp.obj
[ 68%] Building CXX object common/CMakeFiles/common.dir/shortyuv.cpp.obj
[ 70%] Building CXX object common/CMakeFiles/common.dir/picyuv.cpp.obj
[ 72%] Building CXX object encoder/CMakeFiles/encoder.dir/weightPrediction.cpp.obj
[ 74%] Building CXX object encoder/CMakeFiles/encoder.dir/__/x265-extras.cpp.obj
[ 76%] Building CXX object common/CMakeFiles/common.dir/common.cpp.obj
[ 78%] Building CXX object common/CMakeFiles/common.dir/param.cpp.obj
[ 80%] Building CXX object common/CMakeFiles/common.dir/frame.cpp.obj
[ 82%] Building CXX object common/CMakeFiles/common.dir/framedata.cpp.obj
[ 82%] Built target encoder
[ 84%] Building CXX object common/CMakeFiles/common.dir/cudata.cpp.obj
[ 86%] Building CXX object common/CMakeFiles/common.dir/slice.cpp.obj
[ 88%] Building CXX object common/CMakeFiles/common.dir/lowres.cpp.obj
[ 90%] Building CXX object common/CMakeFiles/common.dir/piclist.cpp.obj
[ 92%] Building CXX object common/CMakeFiles/common.dir/predict.cpp.obj
[ 94%] Building CXX object common/CMakeFiles/common.dir/scalinglist.cpp.obj
[ 96%] Building CXX object common/CMakeFiles/common.dir/quant.cpp.obj
[ 98%] Building CXX object common/CMakeFiles/common.dir/deblock.cpp.obj
[ 98%] Built target common
Scanning dependencies of target x265-static
[100%] Linking CXX static library libx265.a
[100%] Built target x265-static
doing cmake in /home/zplovani/sandbox/win32/x265_high_bitdepth_10/8bit with PATH=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH with extra_args=-DENABLE_SHARED=OFF -DHIGH_BIT_DEPTH=ON -DENABLE_ASSEMBLY=OFF -DWINXP_SUPPORT:BOOL=TRUE ../source -DENABLE_ASSEMBLY:BOOL=ON -DEXPORT_C_API:BOOL=ON -DENABLE_CLI:BOOL=ON -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT:BOOL=ON -DEXTRA_LIB=libx265_main10.a like this:
cmake -GUnix Makefiles . -DENABLE_STATIC_RUNTIME=1 -DCMAKE_SYSTEM_NAME=Windows -DCMAKE_RANLIB=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-ranlib -DCMAKE_C_COMPILER=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-gcc -DCMAKE_CXX_COMPILER=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++ -DCMAKE_RC_COMPILER=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-windres -DCMAKE_INSTALL_PREFIX=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32 -DENABLE_SHARED=OFF -DHIGH_BIT_DEPTH=ON -DENABLE_ASSEMBLY=OFF -DWINXP_SUPPORT:BOOL=TRUE ../source -DENABLE_ASSEMBLY:BOOL=ON -DEXPORT_C_API:BOOL=ON -DENABLE_CLI:BOOL=ON -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT:BOOL=ON -DEXTRA_LIB=libx265_main10.a
-- cmake version 3.6.2
-- The C compiler identification is GNU 7.1.0
-- The CXX compiler identification is GNU 7.1.0
-- Check for working C compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-gcc
-- Check for working C compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++
-- Check for working CXX compiler: /home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detected x86 target processor
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Performing Test CC_HAS_NO_STRICT_OVERFLOW
-- Performing Test CC_HAS_NO_STRICT_OVERFLOW - Success
-- Performing Test CC_HAS_NO_NARROWING
-- Performing Test CC_HAS_NO_NARROWING - Success
-- Performing Test CC_HAS_NO_ARRAY_BOUNDS
-- Performing Test CC_HAS_NO_ARRAY_BOUNDS - Success
-- Performing Test CC_HAS_FAST_MATH
-- Performing Test CC_HAS_FAST_MATH - Success
-- Performing Test CC_HAS_STACK_REALIGN
-- Performing Test CC_HAS_STACK_REALIGN - Success
-- Performing Test CC_HAS_FNO_EXCEPTIONS_FLAG
-- Performing Test CC_HAS_FNO_EXCEPTIONS_FLAG - Success
-- Found yasm: /usr/bin/yasm.exe (found version "1.3.0")
-- Found Yasm 1.3.0 to build assembly primitives
-- hg found at /usr/bin/hg
-- x265 version 2.5+6-d11482e5fedb
-- The ASM_YASM compiler identification is unknown
-- Found assembler: /usr/bin/yasm.exe
-- Looking for strtok_r
-- Looking for strtok_r - found
-- Looking for include file getopt.h
-- Looking for include file getopt.h - found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    ENABLE_STATIC_RUNTIME


-- Build files have been written to: /home/zplovani/sandbox/win32/x265_high_bitdepth_10/8bit

making /home/zplovani/sandbox/win32/x265_high_bitdepth_10/8bit as $ PATH=/home/zplovani/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH make  -j 4

Scanning dependencies of target encoder
Scanning dependencies of target common
[  2%] Building CXX object encoder/CMakeFiles/encoder.dir/analysis.cpp.obj[  2%]
Building CXX object encoder/CMakeFiles/encoder.dir/search.cpp.obj
[  3%] Building CXX object encoder/CMakeFiles/encoder.dir/bitcost.cpp.obj
[  4%] Building ASM_YASM object common/CMakeFiles/common.dir/x86/pixel-a.asm.obj
[  6%] Building CXX object encoder/CMakeFiles/encoder.dir/motion.cpp.obj
[  7%] Building CXX object encoder/CMakeFiles/encoder.dir/slicetype.cpp.obj
[  8%] Building ASM_YASM object common/CMakeFiles/common.dir/x86/const-a.asm.obj
[  9%] Building ASM_YASM object common/CMakeFiles/common.dir/x86/cpu-a.asm.obj
[ 10%] Building ASM_YASM object common/CMakeFiles/common.dir/x86/ssd-a.asm.obj
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/ssd-a.asm:146: error: undefined symbol `m8' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/ssd-a.asm:146: error:  (Each undefined symbol is reported only once.)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/ssd-a.asm:181: error: undefined symbol `m9' (first use)
make[2]: *** [common/CMakeFiles/common.dir/build.make:111: common/CMakeFiles/common.dir/x86/ssd-a.asm.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 12%] Building CXX object encoder/CMakeFiles/encoder.dir/frameencoder.cpp.obj
[ 13%] Building CXX object encoder/CMakeFiles/encoder.dir/framefilter.cpp.obj
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13872: error: undefined symbol `r7' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13872: error:  (Each undefined symbol is reported only once.)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13873: error: undefined symbol `xmmm8' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13874: error: undefined symbol `xmmm11' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13874: error: undefined symbol `xmmm9' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13876: error: undefined symbol `m8' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13876: error: undefined symbol `m9' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13930: error: undefined symbol `m11' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13949: error: undefined symbol `m10' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13995: error: (HADDD:1) undefined symbol `sizeofm10' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13995: error: (HADDD:8) undefined symbol `sizeofm10' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:13997: error: undefined symbol `xmmm10' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14094: error: undefined symbol `m12' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14142: error: (HADDD:1) undefined symbol `sizeofm12' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14142: error: (HADDD:8) undefined symbol `sizeofm12' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14190: error: (HADDD:1) undefined symbol `sizeofm10' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14190: error: (HADDD:8) undefined symbol `sizeofm10' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14191: error: undefined symbol `xmmm12' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14279: error: undefined symbol `m13' (first use)
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14327: error: (HADDD:1) undefined symbol `sizeofm13' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14327: error: (HADDD:8) undefined symbol `sizeofm13' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14375: error: (HADDD:1) undefined symbol `sizeofm12' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14375: error: (HADDD:8) undefined symbol `sizeofm12' in preprocessor
/home/zplovani/sandbox/win32/x265_high_bitdepth_10/source/common/x86/pixel-a.asm:14376: error: undefined symbol `xmmm13' (first use)
make[2]: *** [common/CMakeFiles/common.dir/build.make:63: common/CMakeFiles/common.dir/x86/pixel-a.asm.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:248: common/CMakeFiles/common.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 14%] Building CXX object encoder/CMakeFiles/encoder.dir/level.cpp.obj
[ 15%] Building CXX object encoder/CMakeFiles/encoder.dir/nal.cpp.obj
[ 17%] Building CXX object encoder/CMakeFiles/encoder.dir/sei.cpp.obj
[ 18%] Building CXX object encoder/CMakeFiles/encoder.dir/sao.cpp.obj
[ 19%] Building CXX object encoder/CMakeFiles/encoder.dir/entropy.cpp.obj
[ 20%] Building CXX object encoder/CMakeFiles/encoder.dir/dpb.cpp.obj
[ 21%] Building CXX object encoder/CMakeFiles/encoder.dir/ratecontrol.cpp.obj
[ 23%] Building CXX object encoder/CMakeFiles/encoder.dir/reference.cpp.obj
[ 24%] Building CXX object encoder/CMakeFiles/encoder.dir/encoder.cpp.obj
[ 25%] Building CXX object encoder/CMakeFiles/encoder.dir/api.cpp.obj
[ 26%] Building CXX object encoder/CMakeFiles/encoder.dir/weightPrediction.cpp.obj
[ 28%] Building CXX object encoder/CMakeFiles/encoder.dir/__/x265-extras.cpp.obj
[ 28%] Built target encoder
make: *** [Makefile:128: all] Error 2

**Comment by rdp** [comment]
OK for 32-bit I believe I may have run into that and overcame it here once: https://github.com/rdp/ffmpeg-windows-build-helpers/commit/082b686e37378a0c0f4f300a3da87b61ecb727c8

**Comment by zplov** [comment]
Yeah, that solved it.  Need "-DENABLE_ASSEMBLY=OFF" in the cmake params for both the 8bit and 10bit.

Thanks!

**Comment by kgp700** [comment]
Build sucess with DENABLE_ASSEMBLY=OFF
Thanks:)

**Comment by konyzulphrea** [comment]
Gentlemen, I think you can study my edited script.

However ,these advanced solution, 64-bit environment has higher priority as people keep stacking many and many. however, the decoder may no need 64-bit environment. May be these people are lazy or having other reasons...

**Comment by shivSD** [comment]
How do you build/compile x265 for 10 bit .. is there any build configuration file that needs to be changed for 10 bit ?

**Comment by konyzulphrea** [comment]
That's a good question.

I have made an instruction to let people a try but I try not to use x265
because of operational cost

You can try this by studying my construction below (build_x265)

https://github.com/konyzulphrea/ffmpeg-windows-build-helpers/blob/x265ASMOn_HIGHONLY/crossbuild_20170904_x265ASMON_HIGH.sh

This ensures Doing right.  I may consider to put 8-bit x265 back, I try not
to.




I wonder if there is a way to put 8-bit x264 and 10-bit x264 together.

Kenneth
26 october 2017


2017年10月26日 上午4:45，"shivSD" <notifications@github.com>寫道：

**Comment by rdp** [comment]
these days I think the two "bit depths" are mixed in together FWIW...
